src/modules/claims/claims.service.ts

```diff
@@ -683,11 +685,11 @@ export class ClaimsService {
 
     if (registrationTypes.includes(RegistrationTypes.OnChain)) {
       const { claimType: role, claimTypeVersion: version } = claim;
-      const expiry = expirationTimestamp || eternityTimestamp;
+      const expiry = Math.floor((expirationTimestamp || eternityTimestamp)/1000);
```

Comment:
I think there is no need to divide the `eternityTimestamp` by 1000. It should still be the maximum value, IMO
```suggestion
      const expiry = Math.floor(expirationTimestamp / 1000) || eternityTimestamp;
```